app/actions/index.js

```diff
@@ -224,3 +226,31 @@ export const dropFolder = folder => async dispatch => {
   if (!isDirectory) return
   addDat({ path: folder.path })(dispatch)
 }
+
+export const activateTitleEditing = title => ({
+  type: 'ACTIVATE_TITLE_EDITING',
+  title
+})
+
+export const updateTemporaryTitleValue = title => ({
+  type: 'UPDATE_TEMPORARY_TITLE_VALUE',
+  title
+})
+
+export const updateTitle = (key, path, editValue) => {
+  const filePath = `${path}/dat.json`
+  readFile(filePath)
```

Comment:
should use `async/await`

Reply:
also this fails if there's no `dat.json` yet

Reply:
A. Currently it fails silently (no catch handler) and it returns the function before the writing is finished. Not sure if writing this async would look so cool:

```javascript
(async function () {
   const metadata = JSON.parse(await readFile(filePath)
   await writeFile(filePath, JSON.stringify({... metadata, title: editValue})
})()
```
Not soo much better, but I agree: slightly.

B. I didn't mention the fail of missing `dat.json` because in practice the duration of not-having a  `dat.json` is very short. (milliseconds?) - quicker than any user can edit a title. Also I am not sure how the whole functionality should work: If there is no `dat.json` - can you edit the title? What if the dat.json was deleted while editing or we had an write error? Should there be an alert? Shouldn't this be part of the "lock" discussion (-> future work)?

Reply:
I thought about it failing in the way if someone deletes the file in middle of editing, what will happen. But was struggling with design on how and where to show the alert, maybe alert box like when we delete the dat. Should we create an issue or add it to todo list?